test/PushPullTests.cpp

```diff
@@ -0,0 +1,460 @@
+   auto exampleData = std::string("test data");
+   FirePointers(numPushs,
+                numShots,
+                numPulls,
+                location,
+                exampleData);
+   LOG(DEBUG) << "FireManyPointers_OneToMany finished";
+}
+TEST_F(PushPullTests, FireManyPointers_ManyToOne) {
+   LOG(DEBUG) << "FireManyPointers_ManyToOne";
+   auto const numPushs = 4;
+   auto const numShots = 16;
+   auto const numPulls = 1;
+   auto testIpcFile = std::string("FireManyPointers_ManyToOne");
+   auto const location = std::string("ipc:///tmp/" + testIpcFile);
+   auto exampleData = std::string("test data");
+   FirePointers(numPushs,
+                numShots,
+                numPulls,
+                location,
+                exampleData);
+   LOG(DEBUG) << "FireManyPointers_ManyToOne finished";
+}
+TEST_F(PushPullTests, FireOneVector_OneToOne) {
+   LOG(DEBUG) << "FireOneVector_OneToOne";
+   auto const numPushs = 1;
+   auto const numShots = 1;
+   auto const numPulls = 1;
+   auto testIpcFile = std::string("FireOneVector_OneToOne");
+   auto const location = std::string("ipc:///tmp/" + testIpcFile);
+   std::vector<std::pair<void*, unsigned int>> exampleData;
+   auto vectorData = std::pair<void*, unsigned int>(const_cast<int*>(&numPushs),
+                                                    numShots);
+   exampleData.push_back(vectorData);
+   exampleData.push_back(vectorData);
+   exampleData.push_back(vectorData);
+   FireVector(numPushs,
+              numShots,
+              numPulls,
+              location,
+              exampleData);
+   LOG(DEBUG) << "FireOneVector_OneToOne finished";
+}
+TEST_F(PushPullTests, FireManyVectors_OneToMany) {
+   LOG(DEBUG) << "FireManyVectors_OneToMany";
+   auto const numPushs = 1;
+   auto const numShots = 16;
+   auto const numPulls = 4;
+   auto testIpcFile = std::string("FireManyVectors_OneToMany");
+   auto const location = std::string("ipc:///tmp/" + testIpcFile);
+   std::vector<std::pair<void*, unsigned int>> exampleData;
+   auto vectorData = std::pair<void*, unsigned int>(const_cast<int*>(&numPushs),
+                                                    numShots);
+   exampleData.push_back(vectorData);
+   FireVector(numPushs,
+              numShots,
+              numPulls,
+              location,
+              exampleData);
+   LOG(DEBUG) << "FireManyVectors_OneToMany finished";
+}
+TEST_F(PushPullTests, FireManyVectors_ManyToOne) {
+   LOG(DEBUG) << "FireManyVectors_ManyToOne";
+   auto const numPushs = 4;
+   auto const numShots = 16;
+   auto const numPulls = 1;
+   auto testIpcFile = std::string("FireManyVectors_ManyToOne");
+   auto const location = std::string("ipc:///tmp/" + testIpcFile);
+   std::vector<std::pair<void*, unsigned int>> exampleData;
+   auto vectorData = std::pair<void*, unsigned int>(const_cast<int*>(&numPushs),
+                                                    numShots);
+   exampleData.push_back(vectorData);
```

Comment:
I like the makeup of your tests. 

Just as a remainder: When we decide to make nanomsg work across processes then we need some special magic or test processes that can test cross-process communication. We don't have such tests at the moment so it'll be interesting to see how we can solve that.